sdcm/utils/common.py

```diff
@@ -2927,8 +2928,13 @@ def get_partition_keys(ks_cf: str, session, pk_name: str = 'pk', limit: int = No
     cmd = f'select distinct {pk_name} from {ks_cf}'
     if limit:
         cmd += f' limit {limit}'
-    cql_result = session.execute(cmd)
-    pks_list = [getattr(row, pk_name) for row in cql_result.current_rows]
+    cql_result = session.execute_async(SimpleStatement(
+        cmd,
+        fetch_size=3000)
```

Comment:
@fruch , this code can be used in any longevity. So selected some general reasonable value (in principal the driver should have a similar 'reasonable' value as well)
This query only 'interested' in partitions (not rows-per-partition), so there could be a test with 1M partitions (and 1 row-per-partition) or as in 4d-large-partitions: 60 partitions (+10M rows-per-partition). so it might be 2 different challenges for this query to overcome.

Reply:
> @fruch , this code can be used in any longevity. So selected some general reasonable value (in principal the driver should have a similar 'reasonable' value as well)
> This query only 'interested' in partitions (not rows-per-partition), so there could be a test with 1M partitions (and 1 row-per-partition) or as in 4d-large-partitions: 60 partitions (+10M rows-per-partition). so it might be 2 different challenges for this query to overcome.

The driver defaults to 1000 items per page, but there's no reasonable value here, as you explained totally related to the type of data.

You need to overcome this challenge, otherwise this check is bound to be failing.

Otherwise we can't send it to all longevites, not with testing and validating it's working, on each one this check is activated in.

Reply:
ok, since testing all longevities doesn't seem to be a reasonable effort for this fix, so i'll remove this part and we can only consider change it if encounter a failure of this in the future. The main concern of this PR is anyway not about this part but for the other part, fixing https://github.com/scylladb/scylla-cluster-tests/issues/6146